### ISSUE TITLE
有关横向神经网络预测的问题

### ISSUE THREAD

**Issue by oceanqdu** [FederatedAI/FATE#4389]
场景：我使用机器A 和 机器 B模拟了两个参与方使用横向神经网络模块共同训练得到了一个神经网络模型。
在预测推理阶段，我希望各方可以单独使用模型来推理自己的数据，我看到fate的dsl提供了此功能。
但是我在实现的过程中出现了一个错误。
下边是我在预测阶段的执行过程：
1.首先我对模型进行了部署
`flow model deploy --model-id arbiter-9999#guest-9999#host-10000#model --model-version 202210211134322091730`
2.其次我修改了预测的conf文件并将参与方只保留了9999，因为我希望参与方9999可以单独使用模型进行推理。
`{
    "dsl_version": 2,
    "initiator": {
        "role": "guest",
        "party_id": 9999
    },
    "role": {
        "guest": [
            9999
        ]
    },
    "job_parameters": {
        "common": {
            "model_id": "arbiter-9999#guest-9999#host-10000#model",
            "model_version": "202210211807526850880",
            "job_type": "predict"
        }
    },
    "component_parameters": {
        "role": {
            "guest": {
                "0": {
                    "reader_0": {
                        "table": {
                            "name": "breast_9999",
                            "namespace": "experiment"
                        }
                    }
                }
            }
        }
    }
}`
结果提示成功，但是在fateboard里却出现了一个错误
![11fadbe9c6314de4a8215ba3ee6fdab](https://user-images.githubusercontent.com/49178688/197175790-47ecbdf2-bcb8-4263-bda5-3b940a7c2965.png)
 #之前我在#4330中对横向问题进行了提问，你们的回答是删除其余的role，但是为什么在横向神经网络的推理过程中删除其余的角色会会出现这个问题呢？希望能够得到您的解答。
在给出的所有推理阶段的案例都是多个参与方共同提供数据，但实际情况中的横向联邦学习场景中，在完成训练之后，各个参与方应该能够单独的使用训练所得的模型进行推理预测。在fate中通过横向联邦学习得到模型能够在fate框架之外使用吗？如果可以的话应该怎么使用？

**Comment by oceanqdu** [comment]
同时在使用pipeline中的pipeline.model_convert.convert()方法时同样发生了一个错误。
我的代码如下：
`from pipeline.backend.pipeline import PipeLine
from pipeline.component import Reader, DataTransform, HomoLR, FeatureScale, HeteroSecureBoost, Evaluation, HomoSecureBoost
from pipeline.interface import Data
import os
import time
import sys


pid=int(sys.argv[1])
prole=sys.argv[2]
gid=int(sys.argv[3])
hid=int(sys.argv[4])
tablename1=sys.argv[5]
tablename2=sys.argv[6]

try:
    alpha=int(sys.argv[7])
    batch_size=int(sys.argv[8])
    learning_rate=int(sys.argv[9])

except:
    alpha= 0.01
    batch_size= -1
    learning_rate= 0.15


outputfilename=sys.argv[10]

guest_table_name=tablename1+'_'+str(gid)
host_table_name=tablename2+'_'+str(hid)



pipeline = PipeLine() \
        .set_initiator(role=prole, party_id=int(pid)) \
        .set_roles(guest=int(gid), host=int(hid), arbiter=int(gid))






reader_0 = Reader(name="reader_0")
# set guest parameter
reader_0.get_party_instance(role='guest', party_id=gid).component_param(
    table={"name": guest_table_name, "namespace": "experiment"})
# set host parameter
reader_0.get_party_instance(role='host', party_id=hid).component_param(
    table={"name":host_table_name, "namespace": "experiment"})

data_transform_0=DataTransform(name="datatransform_0",with_label=True,output_format="dense")

# set guest parameter
data_transform_0.get_party_instance(role='guest', party_id=gid)
data_transform_0.get_party_instance(role='host', party_id=[hid])


scale_0=FeatureScale(name="scale_0")

homolr_0=HomoLR(name="homolr_0",
                penalty= "L2",
                tol=1e-05,
                alpha= alpha,
                optimizer= "sgd",
                batch_size= batch_size,
                learning_rate= learning_rate,
                init_param= {
                    "init_method": "zeros"
                },
                max_iter= 30,
                early_stop= "diff",
                encrypt_param= {
                    "method": None
                },
                cv_param= {
                    "n_splits": 4,
                    "shuffle": True,
                    "random_seed": 33,
                    "need_cv": False
                },
                decay= 1,
                decay_sqrt= True
                )


evaluation_0 = Evaluation(name="evaluation_0", eval_type="binary")



pipeline.add_component(reader_0)
pipeline.add_component(data_transform_0, data=Data(data=reader_0.output.data))
pipeline.add_component(scale_0, data=Data(data=data_transform_0.output.data))
pipeline.add_component(homolr_0, data=Data(train_data=scale_0.output.data))
pipeline.add_component(evaluation_0, data=Data(data=homolr_0.output.data))
pipeline.compile()


pipeline.fit()

pipeline.model_convert.convert()


`

报错如下：
![image](https://user-images.githubusercontent.com/49178688/197317737-e0e29a75-e1c8-4043-ba29-2ea3cdbaef3c.png)
请问是什么问题呢？
